### PR TITLE
Add Wyvern Keyboard

### PR DESCRIPTION
**Pull request by mysticworks**
## Description
Adds the PCB for the Wyvern keyboard with default and VIA keymaps. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mysticworks/wyvern/config.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device Descriptor Parameter */
+#define VENDOR_ID       0x4D57
+#define PRODUCT_ID      0x0001
```
Comment (drashna) [inline keyboards/mysticworks/wyvern/config.h:23]: This might be better. 
```suggestion
#define PRODUCT_ID      0x39E8
```

Comment (mysticworks) [reply keyboards/mysticworks/wyvern/config.h:23]: Could you explain why this would be better?

Comment (mysticworks) [reply keyboards/mysticworks/wyvern/config.h:23]: If this is about the Vendor ID not being unique, just pushed a change to fix that.

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/mysticworks/wyvern/rules.mk [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (zvecr) [inline keyboards/mysticworks/wyvern/rules.mk:21]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/mysticworks/wyvern/config.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* GRAVE_ESC will always act as ESC when CTRL is held. */
+#define GRAVE_ESC_CTRL_OVERRIDE
```
Comment (zvecr) [inline keyboards/mysticworks/wyvern/config.h:49]: This should be removed, and added to a `keyboards/mysticworks/wyvern/keymaps/default/config.h` if required.

File: keyboards/mysticworks/wyvern/info.json [hunk 4]
```diff
@@ -0,0 +1,851 @@
...
+                {"x":6.75, "y":4, "w":1.25},
+                {"x":8, "y":4, "w":6.25},
+                {"x":14.25, "y":4, "w":1.5},
+                {"x":15.75, "y":4, "w":1.5},
+                {"x":17.5, "y":4.25},
+                {"x":18.5, "y":4.25},
+                {"x":19.5, "y":4.25}
+            ]
+        },
+        "LAYOUT_iso_numpad_left_split_bs": {
```
Comment (zvecr) [inline keyboards/mysticworks/wyvern/info.json:106]: The following layouts do not exist within `wyvern.h`, and should either be implemented or removed.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/mysticworks/wyvern/info.json (hunk 4 above)
Comment (noroadsleft) [reply keyboards/mysticworks/wyvern/info.json:106]: For the record, I would be okay with merging this PR without this change, and having to personally come back around to add that later.

**Comment by mysticworks** [comment]
Requested changes have all been implemented!

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Review by fauxpark — APPROVED** [review]

File: keyboards/mysticworks/wyvern/wyvern.h [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    { k10, k11, k12, k13, k14, k15, k16, k17, XXX, k19 }, \
+    { k20, k21, k22, k23, k24, k25, k26, k27, k28, XXX }, \
+    { k30, k31, k32, k33, k34, k35, k36, k37, XXX, k39 }, \
+    { XXX, k41, k42, k43, k44, k45, k46, k47, XXX, k49 }, \
+    { k50, k51, k52, k53, k54, k55, k56, k57, k58, k59 }, \
+    { k60, k61, k62, k63, k64, k65, k66, k67, k68, k69 }, \
+    { k70, k71, k72, k73, k74, k75, k76, k77, XXX, k79 }, \
+    { XXX, XXX, k82, k83, XXX, k85, XXX, k87, XXX, k89 }, \
+    { k90, k91, k92, XXX, XXX, XXX, XXX, k97, k98, k99 }  \
+}
```
Comment (fauxpark) [inline keyboards/mysticworks/wyvern/wyvern.h:129]: There needs to be a newline at the end of this file.
```
keyboards/mysticworks/wyvern/wyvern.h:111:33: error: backslash-newline at end of file [-Werror]
```

Comment (mysticworks) [reply keyboards/mysticworks/wyvern/wyvern.h]: Fixed

**Comment by fauxpark** [comment]
Thanks!

**Pull request merged**